src/python_testing/TC_SEAR_1_2.py

```diff
@@ -0,0 +1,384 @@
+                    asserts.assert_true(a.positionTag is NullValue or a.positionTag in range(0, self.MAX_RELPOS_ID),
+                             f"SupportedAreas entry with AreaID({a.areaID}) has invalid PositionTag({a.positionTag})")
+        #save so other methods can use this if neeeded
+        self.areaid_list = areaid_list
+
+    async def read_and_validate_selected_areas(self, step):
+        self.print_step(step, "Read SelectedAreas attribute")
+        selected_areas = await self.read_sear_attribute_expect_success(
+            endpoint=self.endpoint, attribute=Clusters.ServiceArea.Attributes.SelectedAreas)
+        logging.info(f"SelectedAreas {selected_areas}")
+
+        #TODO how to check if all entries are uint32?
+
+        asserts.assert_true(len(selected_areas) <= len(self.areaid_list),
+                             f"SelectedAreas(len {len(selected_areas)}) should have at most {len(self.areaid_list)} entries")
+
+        selareaid_list = []
+        for a in selected_areas:
+            if a.areaID in selareaid_list:
+                asserts.fail("SelectedAreas must have unique AreaID values!")
+            else:
+                selareaid_list.append(a.areaID)
+                asserts.assert_true(a.areaID in self.areaid_list,
+                             f"SelectedAreas entry {a.areaID} has invalid value")
+        #save so other methods can use this if neeeded
+        self.selareaid_list = selareaid_list
+
+    async def read_and_validate_current_area(self, step):
+        self.print_step(step, "Read CurrentArea attribute")
+        current_area = await self.read_sear_attribute_expect_success(
+            endpoint=self.endpoint, attribute=Clusters.ServiceArea.Attributes.CurrentArea)
+        logging.info(f"CurrentArea {current_area}")
+
+        asserts.assert_true((len(self.selareaid_list) == 0 and current_area is NullValue)
+                             or
+                             current_area in self.selareaid_list,
+                             f"CurrentArea {current_area} is invalid. SelectedAreas is {self.selareaid_list}.")
+        #save so other methods can use this if neeeded
+        self.current_area = current_area
+        
+    async def read_and_validate_estimated_end_time(self, step):
+        import time
+        read_time = int(time.time())
+        self.print_step(step, "Read EstimatedEndTime attribute")
+        estimated_end_time = await self.read_sear_attribute_expect_success(
+            endpoint=self.endpoint, attribute=Clusters.ServiceArea.Attributes.EstimatedEndTime)
+        logging.info(f"EstimatedEndTime {estimated_end_time}")
+
+        if self.current_area is NullValue:
+            asserts.assert_true(estimated_end_time is NullValue,
+                             "EstimatedEndTime should be null if CurrentArea is null.")
+        else:
+            #allow for some clock skew
+            asserts.assert_true(estimated_end_time >= read_time - 3*60,
```

Comment:
what's the ` - 3*60` doing here?

Reply:
see the code comment above - the device may have a clock skew. perhaps 3 minutes is excessive, so we can lower that

Reply:
https://github.com/CHIP-Specifications/chip-test-plans/pull/4364